File: Week Assignment 5 using C/2_Implement_Stack_Using_LinkedList.c
#include <stdio.h>
#include <stdlib.h>
typedef struct Node{int val;struct Node*next;}Node;
typedef struct{Node*head;}Stack;
Stack*createStack(){Stack*s=malloc(sizeof(Stack));s->head=NULL;return s;}
int isEmpty(Stack*s){return s->head==NULL;}
void push(Stack*s,int x){Node*n=malloc(sizeof(Node));n->val=x;n->next=s->head;s->head=n;}
int pop(Stack*s){if(isEmpty(s)){printf("Underflow\n");return -1;}Node*t=s->head;int v=t->val;s->head=t->next;free(t);return v;}
int peek(Stack*s){return isEmpty(s)?-1:s->head->val;}
int main(){Stack*s=createStack();push(s,1);push(s,2);printf("%d\n",peek(s));printf("%d\n",pop(s));return 0;}

File: Week Assignment 5 using Python/2_Implement_Stack_Using_LinkedList.py
class Node:
    def __init__(self,v,nxt=None): self.v=v; self.next=nxt
class LinkedStack:
    def __init__(self): self.head=None
    def push(self,x): self.head=Node(x,self.head)
    def pop(self):
        if not self.head: raise IndexError
        v=self.head.v; self.head=self.head.next; return v
    def peek(self): return None if not self.head else self.head.v
if __name__=='__main__': s=LinkedStack(); s.push(10); s.push(20); print(s.peek()); print(s.pop())